src/riak_kv_ts_api.erl

```diff
@@ -106,10 +106,18 @@ put_data(Data, Table, Mod) ->
     end.
 
 put_data_to_partitions(Data, Bucket, BucketProps, DDL, Mod) ->
+    DDL = Mod:get_ddl(),
+    BucketProps = riak_core_bucket:get_bucket(Bucket),
     NVal = proplists:get_value(n_val, BucketProps),
+
     PartitionedData = partition_data(Data, Bucket, BucketProps, DDL, Mod),
     PreflistData = add_preflists(PartitionedData, NVal,
                                  riak_core_node_watcher:nodes(riak_kv)),
+
+    SendFullBatches = riak_core_capability:get({riak_kv, w1c_batch_vnode}, false),
```

Comment:
sigh. In a commit that also apparently didn't make it in. Good catch, need to dig deeper.